src/plugins/intel_cpu/src/graph.cpp

```diff
@@ -450,6 +450,37 @@ void Graph::ResolveEdgeConflicts() {
         InsertReorder(edge, layerName, edge->getInputDesc(), edge->getOutputDesc(), isOptimized);
     };
 
+    auto reuseReorderForReadonlyChildren = [&](EdgePtr& edge) {
```

Comment:
Can we have it as an optimization in scope of ApplyImplSpecificGraphOptimizations path?
I believe in this case we don't impact Mergetransposeandreorder optimization.
So, we perform Mergetransposeandreorder first and then we perform reorders optimization.
Of visa versa, depending on the potential benefits.

Reply:
> Can we have it as an optimization in scope of ApplyImplSpecificGraphOptimizations path?
I believe in this case we don't impact Mergetransposeandreorder optimization. So, we perform MergeTransposeAndReorder first and then we perform reorders optimization.

One purpose of this PR is to enable @jane-intel 's https://github.com/openvinotoolkit/openvino/pull/18775  which means there will be multiple children reorders from same transpose node (with her PR), and your expectation `don't impact Mergetransposeandreorder` will be disappointed. we have to change `MergeTransposeAndReorder` to allow such topology anyway.

>If the problem is that we do not want to touch Reorders which resolve in-place conflicts, can we have that as a path, i.e. ResolveInPlaceConflicts() and run it after the optimization you have introduced?

If we take a close look at `Graph::ResolveEdgeConflicts`, we can recognize that it's actually composed of two transformation passes just like any other pass inside `optimizer.ApplyImplSpecificGraphOptimizations` (although it's name didn't suggest that):
 - first pass : checking every pair of parent-child and insert reorder if their memory expectations are inconsistent.
 - second pass: resolve inplace conflict.

my change is just adding another optimization into first pass : reusing existing reorder.

@dmitry-gorokhov & you both expressed same concern about this change, because you didn't consider `ResolveEdgeConflicts` as a pass, just like any other passes inside `ApplyImplSpecificGraphOptimizations`, but to me `ResolveEdgeConflicts` is a pass, we can even split it into 2 function and move them into `ApplyImplSpecificGraphOptimizations` if we like.

Reply:
From my opinion, even this might be a more verbose change / implementation, it follows the general principle of the transformation pipeline - do one thing at a time.
This approach is simply more scalable and testable.
As a theoretical example, we might want to start testing this `ShareReorders` logic.
And to isolate the logic, we could create a graph with reorders explicitly inserted into the outputs of the same port.
And this would be impossible to test your optimization without executing `ResolveEdgeConflicts` call, when the idea of the test is not to test resolving of the edge conflicts, but `ShareReorders`
Regarding `ResolveEdgeConflicts`, I agree it is very similar to the graph optimizations paths. The only difference is it is actually mandatory (so we cannot really call it "optimization").
So, we can consider the whole scenario as:
1) run mandatory path first `ResolveEdgeConflicts` (after we split it of course)
2) optimize / clean up the graph after using optimization path `ShareReorders`

Reply:
OK, reasonable. I changed according to your suggestion.